<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1180</width>
    <height>675</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>515</height>
   </size>
  </property>
  <property name="windowTitle">
   <string notr="true">FreeJoy QtConfigurator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="layoutG_MainLayout" rowstretch="10,1" columnstretch="1,5">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Device</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_7" rowstretch="0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0">
       <property name="leftMargin">
        <number>6</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>6</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <property name="verticalSpacing">
        <number>9</number>
       </property>
       <item row="12" column="2">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="2">
        <widget class="QComboBox" name="comboBox_HidDeviceList">
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>26</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 22px;</string>
         </property>
        </widget>
       </item>
       <item row="5" column="2">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="9" column="2">
        <widget class="QPushButton" name="pushButton_LoadFromFile">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Load config from file</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="7" column="2">
        <widget class="QPushButton" name="pushButton_SetDefaultConfig">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Set file to default config</string>
         </property>
        </widget>
       </item>
       <item row="6" column="2">
        <widget class="QPushButton" name="pushButton_LoadDefaultConfig">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Load default config</string>
         </property>
        </widget>
       </item>
       <item row="8" column="2">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="13" column="2">
        <widget class="QPushButton" name="pushButton_ResetAllPins">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Reset All</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QPushButton" name="pushButton_ReadConfig">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Read config from Device</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="label_DeviceStatus">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QPushButton" name="pushButton_WriteConfig">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Write config to Device</string>
         </property>
        </widget>
       </item>
       <item row="11" column="2">
        <widget class="QPushButton" name="pushButton_SaveToFile">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Save config to file</string>
         </property>
        </widget>
       </item>
       <item row="14" column="2">
        <widget class="QPushButton" name="pushButton_ShowDebug">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Show debug</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QTabWidget" name="tabWidget">
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QTabWidget::pane
{
border-top: 1px solid rgb(190, 190, 190);
border-left: 1px solid rgb(190, 190, 190);
border-right: 1px solid rgb(190, 190, 190);
border-bottom: 1px solid rgb(190, 190, 190);
}</string>
      </property>
      <property name="currentIndex">
       <number>8</number>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tab_PinConfig">
       <attribute name="title">
        <string> Pin Config </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabPinConfig">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>9</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_ButtonConfig">
       <attribute name="title">
        <string> Button Config </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabButtonConfig">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AxesConfig">
       <attribute name="title">
        <string> Axes Config </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabAxesConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AxesCurves">
       <attribute name="title">
        <string> Axes Curves </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabAxesCurvesConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_ShiftRegisters">
       <attribute name="title">
        <string> Shift Registers </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabShiftRegistersConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_Encoders">
       <attribute name="title">
        <string> Encoders</string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabEncodersConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_LED">
       <attribute name="title">
        <string> LED </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabLedConfig">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AdvancedSettings">
       <attribute name="title">
        <string> Advanced Settings </string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_11">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <property name="spacing">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="AdvancedSettings" name="widget_2" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_Debug">
       <attribute name="title">
        <string> Debug </string>
       </attribute>
       <widget class="QLabel" name="label_DebDeviceName">
        <property name="geometry">
         <rect>
          <x>120</x>
          <y>20</y>
          <width>171</width>
          <height>21</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>TextLabel</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_3">
        <property name="geometry">
         <rect>
          <x>400</x>
          <y>0</y>
          <width>121</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>REPORT_ID_JOY count</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEdit">
        <property name="geometry">
         <rect>
          <x>410</x>
          <y>20</y>
          <width>101</width>
          <height>31</height>
         </rect>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
       </widget>
       <widget class="QLabel" name="label_DebDeviceStatus">
        <property name="geometry">
         <rect>
          <x>600</x>
          <y>30</y>
          <width>161</width>
          <height>51</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Connected</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_2">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>0</y>
          <width>91</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Device name:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_VID">
        <property name="geometry">
         <rect>
          <x>310</x>
          <y>30</y>
          <width>47</width>
          <height>13</height>
         </rect>
        </property>
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_TestButton">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>50</y>
          <width>171</width>
          <height>61</height>
         </rect>
        </property>
        <property name="text">
         <string notr="true">Test button</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_ShowDebug2_2">
        <property name="geometry">
         <rect>
          <x>570</x>
          <y>10</y>
          <width>15</width>
          <height>597</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="styleSheet">
         <string notr="true">min-width: 13px;
max-width: 13px;
padding: 0px;</string>
        </property>
        <property name="text">
         <string>D
e
b
u
g</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <layout class="QVBoxLayout" name="layoutV_DebugWindow">
      <property name="spacing">
       <number>0</number>
      </property>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>AdvancedSettings</class>
   <extends>QWidget</extends>
   <header>widgets/advancedsettings.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
